src/v/config/property.h

```diff
@@ -864,9 +872,14 @@ class retention_duration_property final
   : public property<std::optional<std::chrono::milliseconds>> {
 public:
     using property::property;
+    using property::set_value;
+
     void set_value(std::any v) final {
-        update_value(std::any_cast<std::chrono::milliseconds>(std::move(v)));
+        update_value(
+          std::any_cast<std::optional<std::chrono::milliseconds>>(std::move(v))
+            .value_or(-1ms));
```

Comment:
Was this a required change?

Reply:
It is to make the existing tests work. basically, the pattern could be for a property<T> to apply a any_cast<T>.
Here T is optional<milliseconds> but the any_cast is to another type (milliseconds), so the change is needed so that the templated property::set_value can work correctly for retention_duration_property.

previously when calling retention.set_value(1ms) it would go to retention_duration_property::set_value, but with this pr it goes through the templated property::set_value. it's like Christmas, wrapping and unwrapping values.